skglm/estimators.py

```diff
@@ -153,8 +156,11 @@ def path(self, X, y, alphas, coef_init=None, return_n_iter=True, **params):
             The number of iterations along the path. If return_n_iter is set to `True`.
         """
         path_func = cd_solver_path if y.ndim == 1 else bcd_solver_path
+        penalty = compiled_clone(self.penalty)
+        datafit = compiled_clone(self.datafit, to_float32=X.dtype is np.float32)
```

Comment:
This doesn't test for equality

```python
X = np.ones(5, dtype=np.float32)
X.dtype is np.float32  # False
```
```suggestion
        datafit = compiled_clone(self.datafit, to_float32=X.dtype == np.float32)
```

Hence, ``test_design.py`` threw an error when I added the case of float32